src/storage/storage.cpp

```diff
@@ -189,35 +190,13 @@ int Storage::Run()
         throw util::exception("Could not open " + config.hsgr_data_path.string() + " for reading.");
     }
 
-    util::FingerPrint fingerprint_valid = util::FingerPrint::GetValid();
-    util::FingerPrint fingerprint_loaded;
-    hsgr_input_stream.read((char *)&fingerprint_loaded, sizeof(util::FingerPrint));
-    if (fingerprint_loaded.TestGraphUtil(fingerprint_valid))
-    {
-        util::SimpleLogger().Write(logDEBUG) << "Fingerprint checked out ok";
-    }
-    else
-    {
-        util::SimpleLogger().Write(logWARNING) << ".hsgr was prepared with different build. "
-                                                  "Reprocess to get rid of this warning.";
-    }
-
-    // load checksum
     unsigned checksum = 0;
-    hsgr_input_stream.read((char *)&checksum, sizeof(unsigned));
+    std::size_t number_of_graph_nodes = 0;
+    std::size_t number_of_graph_edges = 0;
+    std::tie(checksum, number_of_graph_nodes, number_of_graph_edges) = io::readHSGRSize(hsgr_input_stream);
```

Comment:
Depends on tuple order - there are no named product types in C++ (think Python's namedtuple) but at least create a small type and use semantic member names here.